lambdaclass/mirra_backend
# [GH-835] Randomize Zone shrinking position

> [!WARNING]  
> Merged alonside this [Front PR](https://github.com/lambdaclass/champions_of_mirra/pull/2028)

## Motivation

In search of improving gameplay and making each match different, we want the zone to have a random center position. In our current state, the zone always shrinks in the center of the map

Closes #835 

## Summary of changes

- Added a shrinking position for the zone
- Added a `zone_random_position_radius` to set a smaller radius / bigger radius for the zone to get a random shrinking center position
- Now we send the shrinking center position to the front
 
DISCLAIMER: Although this PR introduces logic to randomize the zone shrinking position, we're not gonna use it because in the client it seems to be a bit complex to fix the zone vfx. I have achieved this disable state by setting the `zone_random_position_radius` field set to 0.

![image](https://github.com/user-attachments/assets/f2f4d1b8-e559-4710-9638-dd27124c9c56)

## How to test it?

- Run again the seeds or modify the new added fields in the configurator.
- Play a match and see that the zone shrinks in a different position

## Checklist
